messages.go

```diff
@@ -178,6 +188,51 @@ func (db *baseDB) auth(
 	}
 }
 
+// logStartupNotice will handle notice messages during the startup process. It will parse them and log them for the
+// client. Notices are not common and only happen if there is something the client should be aware of. So logging should
+// not be a problem.
+// Notice messages can be seen in startup: https://www.postgresql.org/docs/13/protocol-flow.html
+// Information on the notice message format: https://www.postgresql.org/docs/13/protocol-message-formats.html
+// Note: This is true for earlier versions of PostgreSQL as well, I've just included the latest versions of the docs.
+func (db *baseDB) logStartupNotice(
+	rd *pool.ReaderContext,
+) error {
+	message := make([]string, 0)
+	// Notice messages are null byte delimited key-value pairs. Where the keys are one byte.
+	for {
+		// Read the key byte.
+		fieldType, err := rd.ReadByte()
+		if err != nil {
+			return err
+		}
+
+		// If they key byte (the type of field this data is) is 0 then that means we have reached the end of the notice.
+		// We can break our loop here and throw our message data into the logger.
+		if fieldType == 0 {
+			break
+		}
+
+		// Read until the next null byte to get the data for this field. This does include the null byte at the end of
+		// fieldValue so we will trim it off down below.
+		fieldValue, err := rd.ReadSlice(0)
```

Comment:
Can't you use `readString(rd)` here?

Reply:
I wasn't aware of that method, but yeah if it will read until the next null character then that's exactly what i need.

Reply:
:+1: this should be ready to go once this is fixed